### ISSUE TITLE
Asking for simulation

### ISSUE THREAD

**Issue by cory020330** [openworm/sibernetic#168]
Hello, I am a student from South Korea and I am trying to get the sibernetic by using MAC, but it says that .h files are located in awkward locations. Can you help me with it? Otherwise, can I get a demo simulation video of about 10 minutes of sibernetic? Thank you.

**Comment by skhayrulin** [comment]
@cory020330 hi could you please provide any stack trace or error messages. For running demo simulation you can use simple demo config from config folder info about how to run this you can find in README.md

**Comment by cory020330** [comment]
I tried running sibernetic with Mac and innate terminal of OSX, and I stayed on the top level folder. After that, I tried to commence the "export PYTHON~" but it said that there's no .h file available in the folder. So I ctrl+c ctrl+v the .h header file to the top level folder and commenced the "export PHTHON~" but the error message that there's no needed .h file in the folder kept showing. Thank you for your replying and I really appreciate your help.

**Comment by cory020330** [comment]
igwanghuiui-MacBook-Pro:sibernetic-development apple$ make all -f makefile.OSX
Building file: src/owSignalSimulator.cpp
Invoking: clang C++ Compiler
g++ -O1 -Wall -c -I/opt/local/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7/ -Iinc -framework OpenCL -fmessage-length=0 -MMD -MP -MF"Release/obj/owSignalSimulator.d" -MT"Release/obj/owSignalSimulator.d" -o "Release/obj/owSignalSimulator.o" "src/owSignalSimulator.cpp"
clang: warning: -framework OpenCL: 'linker' input unused [-Wunused-command-line-argument]
In file included from src/owSignalSimulator.cpp:46:
inc/owSignalSimulator.h:41:12: fatal error: 'Python.h' file not found
  #include <Python.h>

this is the error message. Mine says that I don't have Python.h file and I don't know how to get an access to this file... Can you please help?

**Comment by skhayrulin** [comment]
@cory020330 I suppose that possible makefile for mac is too old we need update it I'll try to do this today when be near my mac. Let you now when finish

**Comment by cory020330** [comment]
Thanks. I really appreciate that.

**Comment by skhayrulin** [comment]
I've just add a couple fixes in makefile.OSX so try now to compile

**Comment by cory020330** [comment]
I really thank you for helping me but I don't think this is still working... The following lines are the codes that I used, my Macbook's python version is 2.7.10, and would that be a problem? I think Python.h is missing in the owSignalsimulator but I don't know any deeper... Thank you so much anyway.

igwanghuiui-MacBook-Pro:~ apple$ python --version
Python 2.7.10
igwanghuiui-MacBook-Pro:~ apple$ cd /Users/apple/Downloads/sibernetic-development 
igwanghuiui-MacBook-Pro:sibernetic-development apple$ export PYTHONHEADERDIR=/usr/local/Cellar/python/2.7.10//Python.framework/Headers/
igwanghuiui-MacBook-Pro:sibernetic-development apple$ export PYTHONLIBDIR=/usr/local/lib/python2...
igwanghuiui-MacBook-Pro:sibernetic-development apple$ export PYTHONFRAMEWORKDIR=/usr/local/Frameworks/
igwanghuiui-MacBook-Pro:sibernetic-development apple$ make clean -f makefile.OSX
rm -rf ./Release/obj/owSignalSimulator.o ./Release/obj/owVtkExport.o ./Release/obj/main.o ./Release/obj/owHelper.o ./Release/obj/owOpenCLSolver.o ./Release/obj/owPhysicsFluidSimulator.o ./Release/obj/owWorldSimulation.o ./Release/obj/owNeuronSimulator.o ./Release/obj/owConfigProperty.o ./Release/obj/test/owPhysicTest.o./Release/obj/owSignalSimulator.d ./Release/obj/owVtkExport.d ./Release/obj/main.d ./Release/obj/owHelper.d ./Release/obj/owOpenCLSolver.d ./Release/obj/owPhysicsFluidSimulator.d ./Release/obj/owWorldSimulation.d ./Release/obj/owNeuronSimulator.d ./Release/obj/owConfigProperty.d ./Release/obj/test/owPhysicTest.d ./Release/Sibernetic
 
igwanghuiui-MacBook-Pro:sibernetic-development apple$ make all -f makefile.OSX
Building file: src/owSignalSimulator.cpp
Invoking: clang C++ Compiler
g++ -std=c++14 -O3 -Wall -c -I/opt/local/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7/ -Iinc -framework OpenCL -fmessage-length=0 -MMD -MP -MF"Release/obj/owSignalSimulator.d" -MT"Release/obj/owSignalSimulator.d" -o "Release/obj/owSignalSimulator.o" "src/owSignalSimulator.cpp"
clang: warning: -framework OpenCL: 'linker' input unused [-Wunused-command-line-argument]
In file included from src/owSignalSimulator.cpp:46:
inc/owSignalSimulator.h:41:12: fatal error: 'Python.h' file not found
  #include <Python.h>
           ^~~~~~~~~~
1 error generated.
make: *** [Release/obj/owSignalSimulator.o] Error 1
igwanghuiui-MacBook-Pro:sibernetic-development apple$

**Comment by skhayrulin** [comment]
could you please also provide version of you MacOS

**Comment by cory020330** [comment]
Mac OS Catalina 10.15.6
Do you want me to update it?

**Comment by skhayrulin** [comment]
No, try to replace current makefile.OSX with this [one](https://gist.github.com/skhayrulin/660230b4376f9ea0ba2b056489e5d646). Let me know if it works

**Comment by cory020330** [comment]
I replaced the makefile.OSX with the file that you provided and the following lines were on the terminal.  Unfortunately, it contained the same error message... I don't know the problem is, I'm sorry

Last login: Tue Jul 28 19:19:18 on ttys000
igwanghuiui-MBP:~ apple$ python -V
Python 2.7.10
igwanghuiui-MBP:~ apple$ cd /Users/apple/Desktop/sibernetic-development 
igwanghuiui-MBP:sibernetic-development apple$ export PYTHONHEADERDIR=/usr/local/Cellar/python/2.7.10//Python.framework/Headers/
igwanghuiui-MBP:sibernetic-development apple$ export PYTHONLIBDIR=/usr/local/lib/python2...
igwanghuiui-MBP:sibernetic-development apple$ export PYTHONFRAMEWORKDIR=/usr/local/Frameworks/
igwanghuiui-MBP:sibernetic-development apple$ make clean -f makefile.OSX
rm -rf ./Release/obj/owSignalSimulator.o ./Release/obj/owVtkExport.o ./Release/obj/main.o ./Release/obj/owHelper.o ./Release/obj/owOpenCLSolver.o ./Release/obj/owPhysicsFluidSimulator.o ./Release/obj/owWorldSimulation.o ./Release/obj/owNeuronSimulator.o ./Release/obj/owConfigProperty.o ./Release/obj/test/owPhysicTest.o./Release/obj/owSignalSimulator.d ./Release/obj/owVtkExport.d ./Release/obj/main.d ./Release/obj/owHelper.d ./Release/obj/owOpenCLSolver.d ./Release/obj/owPhysicsFluidSimulator.d ./Release/obj/owWorldSimulation.d ./Release/obj/owNeuronSimulator.d ./Release/obj/owConfigProperty.d ./Release/obj/test/owPhysicTest.d ./Release/Sibernetic
 
igwanghuiui-MBP:sibernetic-development apple$ make all -f makefile.OSX
Building file: src/owSignalSimulator.cpp
Invoking: clang C++ Compiler
g++ -std=c++14 -O3 -Wall -c -I/System/Library/Frameworks/Python.framework/Versions/2.7/Headers/ -Iinc -framework OpenCL -fmessage-length=0 -MMD -MP -MF"Release/obj/owSignalSimulator.d" -MT"Release/obj/owSignalSimulator.d" -o "Release/obj/owSignalSimulator.o" "src/owSignalSimulator.cpp"
clang: warning: -framework OpenCL: 'linker' input unused [-Wunused-command-line-argument]
In file included from src/owSignalSimulator.cpp:46:
inc/owSignalSimulator.h:41:12: fatal error: 'Python.h' file not found
  #include <Python.h>
           ^~~~~~~~~~
1 error generated.
make: *** [Release/obj/owSignalSimulator.o] Error 1
igwanghuiui-MBP:sibernetic-development apple$

**Comment by skhayrulin** [comment]
Ok could you try to build sibernetic in new terminal window and don't redefine `PYTHONLIBDIR, PYTHONHEADERDIR, PYTHONFRAMEWORKDIR`  just run  `make all -f makefile.OSX`. Also could you check if your mac really contains folder ` /System/Library/Frameworks/Python.framework/Versions/2.7/Headers/` and Python.h file in it

**Comment by md-121** [comment]
Hi everyone,

I also got the "'Python.h' file not found" error during installation but using the makefile.OSX @skhayrulin linked solved my problem! Thank you.
I tested it using macOS Big Sur 11.0.1.

Just wanted to let you know. Maybe this will help someone.
